autoPyTorch/datasets/time_series_dataset.py

```diff
@@ -1,174 +1,1267 @@
+    df_range = pd.concat([df_range_end - df_range, df_range_end])
+    value_ranges = df_range[list(queried_features)].T.values
+    feature_index: List[int] = sum([list(range(*value_r)) for value_r in value_ranges], [])
+    feature_index.sort()
+    return tuple(feature_index)  # type: ignore[return-value]
+
+
+def compute_time_features(start_time: pd.DatetimeIndex,
+                          date_period_length: int,
+                          time_feature_length: int,
+                          freq: str,
+                          time_feature_transforms: List[TimeFeature]) -> np.ndarray:
+    date_info = pd.date_range(start=start_time,
+                              periods=date_period_length,
+                              freq=freq)[-time_feature_length:]
+    try:
+        time_features = np.vstack(
+            [transform(date_info).to_numpy(float)
+             if not isinstance(transform, ConstantTransform) else transform(date_info)
+             for transform in time_feature_transforms]
+        ).T
+    except OutOfBoundsDatetime:
+        # This is only a temporal solution TODO consider how to solve this!
+        time_features = np.zeros([time_feature_length, len(time_feature_transforms)])
+    return time_features
+
+
+class TimeSeriesSequence(Dataset):
+    _is_test_set = False
+    is_pre_processed = False
+
     def __init__(self,
-                 target_variables: Tuple[int],
-                 sequence_length: int,
-                 n_steps: int,
-                 train: TIME_SERIES_FORECASTING_INPUT,
-                 val: Optional[TIME_SERIES_FORECASTING_INPUT] = None,
-                 resampling_strategy: Union[CrossValTypes, HoldoutValTypes] = HoldoutValTypes.holdout_validation,
+                 X: Optional[np.ndarray],
+                 Y: np.ndarray,
+                 start_time: Optional[pd.DatetimeIndex] = None,
+                 freq: str = '1Y',
+                 time_feature_transform: List[TimeFeature] = [ConstantTransform],
+                 X_test: Optional[np.ndarray] = None,
+                 Y_test: Optional[np.ndarray] = None,
+                 train_transforms: Optional[torchvision.transforms.Compose] = None,
+                 val_transforms: Optional[torchvision.transforms.Compose] = None,
+                 n_prediction_steps: int = 1,
+                 sp: int = 1,
+                 known_future_features_index: Optional[Tuple[int]] = None,
+                 compute_mase_coefficient_value: bool = True,
+                 time_features: Optional[np.ndarray] = None,
+                 is_test_set: bool = False,
+                 ) -> None:
+        """
+        A dataset representing a time series sequence.
+        Args:
+            X (Optional[np.ndarray]): past features
+            Y (np.ndarray): past targets
+            start_time (Optional[pd.DatetimeIndex]): times of the first timestep of the series
+            freq (str): frequency that the data is sampled
```

Comment:
could you write the description in a new line with indentation? I think it messes with the documentation builder. [example](https://github.com/dengdifan/Auto-PyTorch/blob/4898ca5ef492b465ad1331df2b380b10be070087/autoPyTorch/data/base_feature_validator.py#L69)